### PR TITLE
identity-outbound-auth-metamask initial codebase

### PR DESCRIPTION
**Pull request by NimashaBandara**

### CODE REVIEW

File: components/org.wso2.carbon.identity.metamask.federated.authenticator/src/main/java/org/wso2/carbon/identity/metamask/federated/authenticator/MetamaskAuthenticationConstants.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        }
+
+        @Override
+        public String toString() {
+
+            return String.format("%s - %s", code, message);
+        }
+    }
+
+}
```
Comment (Thisara-Welmilla) [inline components/org.wso2.carbon.identity.metamask.federated.authenticator/src/main/java/org/wso2/carbon/identity/metamask/federated/authenticator/MetamaskAuthenticationConstants.java]: Add a new line here.

**Review by minolih — APPROVED** [review]

File: components/org.wso2.carbon.identity.metamask.federated.authenticator/src/main/java/org/wso2/carbon/identity/metamask/federated/authenticator/MetamaskAuthenticationConstants.java (hunk 1 above)
Comment (minolih) [reply components/org.wso2.carbon.identity.metamask.federated.authenticator/src/main/java/org/wso2/carbon/identity/metamask/federated/authenticator/MetamaskAuthenticationConstants.java]: This was resolved.

File: Metamask-Login-Page/metamask.jsp [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+  ~
+  ~ Unless required by applicable law or agreed to in writing,
+  ~ software distributed under the License is distributed on an
+  ~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  ~ KIND, either express or implied.  See the License for the
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+  --%>
+  
+  <%@ page import="java.io.File" %>
```
Comment (Thisara-Welmilla) [inline Metamask-Login-Page/metamask.jsp:18]: shall we remove leading whitespaces. check line 21 also.

**Comment by NimashaBandara** [comment]
Corrected and commited

File: Metamask-Login-Page/metamask.jsp (hunk 2 above)
Comment (NimashaBandara) [reply Metamask-Login-Page/metamask.jsp:18]: Corrected and commited

File: components/org.wso2.carbon.identity.metamask.federated.authenticator/src/main/java/org/wso2/carbon/identity/metamask/federated/authenticator/MetamaskAuthenticationConstants.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.metamask.federated.authenticator;
+
+public class MetamaskAuthenticationConstants {
```
Comment (JKAUSHALYA) [inline components/org.wso2.carbon.identity.metamask.federated.authenticator/src/main/java/org/wso2/carbon/identity/metamask/federated/authenticator/MetamaskAuthenticationConstants.java:24]: Need a class level doc comment.

Comment (NimashaBandara) [reply components/org.wso2.carbon.identity.metamask.federated.authenticator/src/main/java/org/wso2/carbon/identity/metamask/federated/authenticator/MetamaskAuthenticationConstants.java:24]: added a comment

**Review by JKAUSHALYA — APPROVED** [review]

**Pull request merged**